Assets/scripts/BattleManager.cs

```diff
@@ -107,46 +123,61 @@ private class PlayerBattleData {
 	public void SetPlayerInstrument(int playerNumber, int instrumentID) {
 		Debug.Assert (players.ContainsKey (playerNumber));
 
-	}
+	}
+
 
 	void OnStartBattle(EnterBattleMessage m) {
 		targetLine.enabled = true;
+		divider.GetComponent<MeshRenderer>().enabled = true;
 		Song song = ServiceFactory.Instance.Resolve<Song>();
 		Debug.Assert (players.ContainsKey (m.AttackingUnit.PlayerNumber));
 		Debug.Assert (players.ContainsKey (m.DefendingUnit.PlayerNumber));
 		// Prepare battle data per player
 		attacker = players [m.AttackingUnit.PlayerNumber];
 		defender = players [m.DefendingUnit.PlayerNumber];
+
 		attacker.unit = m.AttackingUnit;
 		defender.unit = m.DefendingUnit;
+
 		attacker.instrumentID = 0; // Edgy synth
 		defender.instrumentID = 1; // Smoother synth
+
 		attacker.battleNotes = song.GetNextBattleNotes (battleMeasures, attacker.instrumentID, attacker.difficulty);
 		defender.battleNotes = song.GetNextBattleNotes (battleMeasures, defender.instrumentID, defender.difficulty);
+
 		attacker.battleNoteStates = new int[attacker.battleNotes.Length];
 		defender.battleNoteStates = new int[defender.battleNotes.Length];
 
-		// Spawn notes:
-		float margin = 3f;
-		float offset = 10f;
-		float spawnHeight = -4f;
-		float spawnDepth = 13f;
-		float stretchFactor = 10.73f;
-		float currentMusicTime = song.playerPosition;
-		foreach (int playerNumber in players.Keys) {
+		List<int> OrderedPlayers = players.Keys.ToList();
+		OrderedPlayers.Sort();
+
+		float PlayerXPos = UNIT_MARGIN;
+
+		float currentMusicTime = song.playerPosition;
+		// Spawn notes:
+		foreach (int playerNumber in OrderedPlayers) {
 			foreach (Note note in players[playerNumber].battleNotes) {
```

Comment:
So like there could be a check here that the player is either the attacker or defender, but we can cross that bridge when the game turns into a MMO.